core/src/main/java/bio/terra/pearl/core/service/workflow/EventService.java

```diff
@@ -104,19 +118,52 @@ public EnrolleeCreationEvent publishEnrolleeCreationEvent(Enrollee enrollee, Por
                 .portalParticipantUser(ppUser)
                 .enrolleeRuleData(enrolleeRuleData)
                 .build();
+        savePublishedEvent(EventClass.ENROLLEE_CREATION_EVENT, ppUser.getPortalEnvironmentId(), enrollee);
         applicationEventPublisher.publishEvent(enrolleeEvent);
         return enrolleeEvent;
     }
 
-    /** adds ruleData to the event, and also ensures the enrollee task list is refreshed */
+    /**
+     * Saves a record of the published event. If the event does not involve a specific enrollee/portalEnv,
+     * they can be set to null.
+     */
```

Comment:
I don't think I mentioned this in the last review, but this was a great choice to make this method private -- that really clearly communicates that saving is handled internally

Reply:
If there are two things I got from my Northeastern education, it's to:
- Never make anything public
- Make so many tests you go dizzy from looking at them